src/app/static/src/app/store/plottingSelections/actions.ts

```diff
@@ -0,0 +1,40 @@
+import {ActionContext, ActionTree} from "vuex";
+import {PlottingSelectionsMutations} from "./mutations";
+import {
+    BarchartSelections, BubblePlotSelections, ChoroplethSelections,
+    PlottingSelectionsState
+} from "./plottingSelections";
+import {PayloadWithType} from "../../types";
+import {DataExplorationState} from "../dataExploration/dataExploration";
+
+export interface PlottingSelectionsActions {
+    updateBarchartSelections: (store: ActionContext<PlottingSelectionsState, DataExplorationState>, payload: PayloadWithType<BarchartSelections>) => void
+    updateChoroplethSelections: (store: ActionContext<PlottingSelectionsState, DataExplorationState>, payload: PayloadWithType<Partial<ChoroplethSelections>>) => void
+    updateBubblePlotSelections: (store: ActionContext<PlottingSelectionsState, DataExplorationState>, payload: PayloadWithType<Partial<BubblePlotSelections>>) => void
+}
+
+export const actions: ActionTree<PlottingSelectionsState, DataExplorationState> & PlottingSelectionsActions = {
```

Comment:
Yeah, I can't think of a better way than this, though it seems quite long winded! If it hadn't been for the Bubble plot anomaly of having two indicator ids, I might have suggested having another plugin to monitor mutations, but since there's only three and one of them's different this is probably the sanest approach.